include/zephyr/drivers/i2c.h

```diff
@@ -352,6 +352,49 @@ typedef int (*i2c_target_read_requested_cb_t)(
 typedef int (*i2c_target_read_processed_cb_t)(
 		struct i2c_target_config *config, uint8_t *val);
 
+#ifdef CONFIG_I2C_TARGET_DMA_MODE
```

Comment:
Sort of in line with my other comments, I'd suggest naming this as it functions rather than how it might be implemented.

E.g. I2C_TARGET_BUFFER_MODE or I2C_TARGET_BATCH_MODE

Reply:
Renamed to I2C_TARGET_BUFFER_MODE.